techmatters/hrm
# Chi 1594: Reindex endpoint

## Description

Review after #369 is merged,

* Adds an static key based 'admin auth' where the key is built using a role name rather than a Twilio account name - this can only access the new endpoint created in this PR
* Adds a 'reindex' endpoint which can be called to initiate reindexing of search documents manually. It queries data from the DB and posts it as messages onto the appropriate SQS queue. Resources can be specified either by a range of updated dates or a list of specific resource IDs. Updated date ranges can either be limited to a single account, or be global across every account in the region / environment. If a resource ID list is specified, it must be limited to a single account, this is because resource IDs uniqueness is only enforced within an account, so resource ID clashes across accounts could result in unexpected behaviour if an account ID were not specified. A limit is not enforced, but data is taken from the DB for post
* Adds some lightweight SQS and SSM mocks that can be run within the test process, so service tests can mock & validate these interactions without the overhead of localstack 

TODO: Infrastructure changes to make this endpoint callable

### Checklist


### Related Issues
CHI-1594

### Verification steps

Once the endpoint is callable:

1. Call the endpoint via postman / curl (or however we expose it)
2. Review the AWS and elasticCloud consoles to ensure the expected resources were indexed in ES
